### PR TITLE
socket.create_connection source_address is optional.

### LINKED ISSUE #4116: socket.create_connection source_address should be optional
```
def create_connection(address: Tuple[Optional[str], int],
                      timeout: Optional[float] = ...,
                      source_address: Tuple[Union[bytearray, bytes, Text], int] = ...) -> socket: ...
```
The `source_address` field should be marked as `Optional`, as it defaults to `None`.

### PR DESCRIPTION
**Pull request by bwelling**
Fixes #4116.

### CODE REVIEW

File: stdlib/2and3/socket.pyi [hunk 1]
```diff
@@ -713,7 +713,7 @@ if sys.version_info >= (3, 7):
 
 def create_connection(address: Tuple[Optional[str], int],
                       timeout: Optional[float] = ...,
-                      source_address: Tuple[Union[bytearray, bytes, Text], int] = ...) -> socket: ...
+                      source_address: Optional[Tuple[Union[bytearray, bytes, Text], int\] = ...) -> socket: ...
```
Comment (srittau) [inline stdlib/2and3/socket.pyi:716]: These seems to be an extraneous backslash here.

Comment (srittau) [reply stdlib/2and3/socket.pyi]: Ah sorry, the backslash was probably supposed to be a closing bracket. So this is missing now. :)

Comment (bwelling) [reply stdlib/2and3/socket.pyi]: Sorry about that.  Should be better now.

**Comment by srittau** [comment]
@bwelling Nearly there. Stubtest fails, because `socket.create_connection` was on its whitelist: `tests/stubtest_whitelists/py3_common.txt`. Since your PR fixes this problem, it needs to removed from there.

**Comment by srittau** [comment]
Travis-CI seems to hiccup, but it passed now: https://travis-ci.org/github/python/typeshed/builds/692261015

**Pull request merged**